subsys/bluetooth/host/conn.c

```diff
@@ -1912,7 +1933,37 @@ int bt_le_set_auto_conn(bt_addr_le_t *addr,
 struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
 					const struct bt_le_adv_param *param)
 {
-	return NULL;
+	int err;
+	struct bt_conn *conn;
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, peer);
+	if (conn) {
+		switch (conn->state) {
+		case BT_CONN_CONNECT_DIR_ADV:
+		case BT_CONN_CONNECT:
+		case BT_CONN_CONNECTED:
+			return conn;
+		default:
+			bt_conn_unref(conn);
+			return NULL;
+		}
+	}
+
+	conn = bt_conn_add_le(peer);
+	if (!conn) {
+		return NULL;
+	}
+
+	err = bt_le_adv_start_internal(param, NULL, 0, NULL, 0, peer);
+	if (err) {
+		BT_WARN("Directed advertising could no be started: %d", err);
+		bt_conn_unref(conn);
+		return NULL;
+	}
+
+	bt_conn_set_state(conn, BT_CONN_CONNECT_DIR_ADV);
```

Comment:
Should we set the state before starting advertising? (in case a connection is established quickly before we reach here?

Reply:
Well, I didn't encounter any problems with it while testing  (even in the test scenario, where high duty advertising is triggered when there is a connect request already pending from central), but I think it is worth changing just for safety even though it would require state clean up in case of an error from ```bt_le_adv_start_internal```

Reply:
ignore this. No action required.